messages/teleporter/message_manager.go

```diff
@@ -93,16 +93,36 @@ func (m *messageManager) GetDestinationBlockchainID(unsignedMessage *warp.Unsign
 	teleporterMessage, err := m.parseTeleporterMessage(unsignedMessage)
 	if err != nil {
 		m.logger.Error(
-			"Failed get teleporter message.",
+			"Failed to parse teleporter message.",
 			zap.String("warpMessageID", unsignedMessage.ID().String()),
 		)
 		return ids.ID{}, err
 	}
+	return teleporterMessage.DestinationBlockchainID, nil
+}
 
-	// Cache the message so it can be reused in SendMessage
-	m.teleporterMessageCache.Put(unsignedMessage.ID(), teleporterMessage)
+func (m *messageManager) GetOriginSenderAddress(unsignedMessage *warp.UnsignedMessage) (common.Address, error) {
+	teleporterMessage, err := m.parseTeleporterMessage(unsignedMessage)
+	if err != nil {
+		m.logger.Error(
+			"Failed to parse teleporter message.",
+			zap.String("warpMessageID", unsignedMessage.ID().String()),
+		)
+		return common.Address{}, err
+	}
+	return teleporterMessage.OriginSenderAddress, nil
+}
 
-	return teleporterMessage.DestinationBlockchainID, nil
+func (m *messageManager) GetDestinationAddress(unsignedMessage *warp.UnsignedMessage) (common.Address, error) {
+	teleporterMessage, err := m.parseTeleporterMessage(unsignedMessage)
```

Comment:
nit: GetDestinationBlockchainID, GetOriginSenderAddress and GetDestinationAddress differ by just the last line as far as I can tell. Should consider abstracting the shared part out to a helper function.

Reply:
These methods already abstract most of their code to `parseTeleporter`. We could remove the logs emitted in GetDestinationBlockchainID, GetOriginSenderAddress and GetDestinationAddress so that they are just the call to `parseTeleporter` and the field access, but I also think there's value in keeping error logs verbose.